src/app/pages/send/send-crypto-asset-form/_components/recipient-field.tsx

```diff
@@ -1,6 +1,29 @@
+import { useState } from 'react';
+
+import { RecipientAccountsDrawer } from './recipient-accounts-drawer/recipient-accounts-drawer';
 import { TextInputField } from './text-input-field';
 
-interface RecipientFieldProps {}
-export function RecipientField({}: RecipientFieldProps) {
-  return <TextInputField name="recipient" label="To" placeholder="Address or name" />;
+interface RecipientFieldProps {
+  onBlur?(): void;
+  topInputOverlay?: JSX.Element;
+}
+export function RecipientField({ onBlur, topInputOverlay }: RecipientFieldProps) {
+  const [isShowing, setIsShowing] = useState(false);
+
+  const toggleAccountsDrawer = () => setIsShowing(!isShowing);
+
+  return (
+    <>
+      <TextInputField
+        name="recipientAddressOrBnsName"
+        label="To"
+        labelAction="Choose account"
+        placeholder="Address or name"
+        onBlur={onBlur}
+        onClickLabelAction={toggleAccountsDrawer}
+        topInputOverlay={topInputOverlay}
+      />
+      <RecipientAccountsDrawer isShowing={isShowing} onCloseAccountsDrawer={toggleAccountsDrawer} />
```

Comment:
Don't think this component belongs here, otherwise it should be named _RecipientFieldAndRecipientAccountDrawer_ component. `<RecipientField />` shouldn't know about all implementation details relating to another piece of functionality.

[Similar to react docs on composition](https://reactjs.org/docs/composition-vs-inheritance.html). `SplitPane` doesn't/shouldn't know about `Contact`/`Chat`.

Why not pass it in as a child? Or make it a route? That way this component remains unaware of this feature, and responsible only for one thing,

Reply:
Sure, I can def change pass it in now or refactor to a route now that the code is migrated over and working, np.